arrow-libs/core/arrow-core-data/src/main/kotlin/arrow/core/Either.kt

```diff
@@ -1588,7 +1588,7 @@ fun <A, B, C> EitherOf<A, B>.ap(ff: EitherOf<A, (B) -> C>): Either<A, C> =
   flatMap { a -> ff.fix().map { f -> f(a) } }
 
 fun <A, B, C> Either<A, B>.apEval(ff: Eval<Either<A, (B) -> C>>): Eval<Either<A, C>> =
-  ff.map { this.ap(it) }
+  fold({ l -> Eval.now(l.left()) }, { r -> ff.map { it.map { f -> f(r) } } })
```

Comment:
is there or should be a specific test for this change?

Reply:
We should add many tests.. 😭😅 This code was taken from the `EitherApply.kt` code, and is verified in `TraverseTest`. By reverting this change it makes `TraverseTest` fail.

Reply:
> By reverting this change it makes TraverseTest fail.

That's what I meant, perfect then! 👌